framework/src/main/java/org/checkerframework/common/wholeprograminference/scenelib/AMethodWrapper.java

```diff
@@ -0,0 +1,117 @@
+     * is vivified by calls to {@link #vivifyParameter(int, String, Name)} or {@link
+     * #vivifyParameter(int, TypeMirror, Name)}.
+     */
+    private Map<Integer, AFieldWrapper> parameters = new HashMap<>();
+
+    /**
+     * The return type, as a fully-qualified name.
+     *
+     * @return the return type as a fully-qualified name, or "java.lang.Object" if the return type
+     *     is unknown.
+     */
+    public @FullyQualifiedName String getReturnType() {
+        return returnType;
+    }
+
+    /**
+     * Provide the AMethodWrapper with a return type.
+     *
+     * @param returnType a fully-qualified name
+     */
+    private void setReturnType(@FullyQualifiedName String returnType) {
+        // do not keep return types that start with a ?, because that's not valid Java
+        if ("java.lang.Object".equals(this.returnType) && !returnType.startsWith("?")) {
+            this.returnType = returnType;
+        }
+    }
+
+    /**
+     * Wrap an AMethod.
+     *
+     * @param theMethod the method to wrap
+     * @param returnType the return type of the method
+     */
+    public AMethodWrapper(AMethod theMethod, TypeMirror returnType) {
+        this.theMethod = theMethod;
+        @SuppressWarnings("signature") // https://tinyurl.com/cfissue/3094
+        @DotSeparatedIdentifiers String typeAsString = returnType.toString();
+        setReturnType(typeAsString);
+    }
+
+    /**
+     * Avoid if possible.
+     *
+     * @return the underlying AMethod object that has been wrapped
+     */
+    public AMethod getAMethod() {
+        return theMethod;
+    }
+
+    /**
+     * Interact with a parameter.
+     *
+     * @param i the parameter index (zero-indexed)
+     * @param type the type of the parameter, as a TypeMirror
+     * @param simpleName the name of the parameter
+     * @return an AFieldWrapper representing the parameter
+     */
+    public AFieldWrapper vivifyParameter(int i, TypeMirror type, Name simpleName) {
+        @SuppressWarnings("signature") // https://tinyurl.com/cfissue/3094
+        @DotSeparatedIdentifiers String typeAsString = type.toString();
+        return vivifyParameter(i, typeAsString, simpleName);
+    }
+
+    /**
+     * Interact with a parameter.
+     *
+     * @param i the parameter index (1st parameter is zero)
+     * @param type a fully-qualified name representing the type of the parameter
+     * @param simpleName the name of the parameter
+     * @return an AFieldWrapper representing the parameter
+     */
+    private AFieldWrapper vivifyParameter(int i, @FullyQualifiedName String type, Name simpleName) {
+        if (parameters.containsKey(i)) {
+            return parameters.get(i);
+        } else {
+            AFieldWrapper wrapper = new AFieldWrapper(theMethod.parameters.getVivify(i), type);
+            parameters.put(i, wrapper);
+            wrapper.setName(simpleName.toString());
```

Comment:
Side-effecting an object is a code smell.  It's better to pass the information to the constructor, so that the object is never in a partially-constructed state.